### DISCUSSION TITLE
Does AMS have any functions or methods to calculate the network traffic metrics or data traffic metrics?

### DISCUSSION THREAD

**Discussion by williehao** [ant-media/Ant-Media-Server#5868] [FAQs]
I have some issues that need your assistance as following:

1.	Does AMS have any functions or methods to calculate the network traffic metrics or data traffic metrics? Azure costs include AMS license, Azure VM, and 'Network traffic cost'. Every cost is fixed except for network traffic cost, so I need to find out how to calculate each AMS user's Internet network traffic metrics(special AMS players ) to charge money from our product customers.

2.	Could I use Swagger (Rest API) on a Azure VM? If yes, how can I achieve this? If not, do you have any suggestions? Alternatively, if I have to create my own Rest API with AMS EE version, what steps should I follow?

![image](https://github.com/ant-media/Ant-Media-Server/assets/15116422/df5afc68-a0eb-4c0a-8c2a-067ae4164dbc)

![image](https://github.com/ant-media/Ant-Media-Server/assets/15116422/f190f4f7-4242-440a-afec-c890acc80650)

**Comment by lastpeony** [comment]
1-) Please check this answer:
https://github.com/orgs/ant-media/discussions/5860#discussioncomment-7701885
2-) You can use rest api any where becuase its in AMS. If you want to implement your own rest api end points you need to utilize plugins feature.

**Comment by williehao** [comment]
@lastpeony  Thank you very much for your prompt feedback. I have one more question: is it impossible to create a Rest API same/similar to AMS (https://antmedia.io/rest/)? If yes, how can I do it, or where can I find the resources?

**Reply by lastpeony** [reply]
You can create your own REST API endpoints using the plugins feature, alongside the options already available in AMS. Please take a look at those documentations for more information.

https://antmedia.io/plugins-will-make-ant-media-server-more-powerful/
https://antmedia.io/plugin-development-guide/
https://antmedia.io/docs/guides/developing-antmedia-server/introduction-plugin-structure/
https://github.com/ant-media/Plugins

If you're discussing the creation of an application server with a REST API, that topic isn't connected to Ant Media Server and im afraid i cannot help you with that.

**Reply by williehao** [reply]
@lastpeony Thanks again, I've tried to clone Swagger API from (https://antmedia.io/rest/) to my EE version environment, but unfortunately ,it isn't working for me and  I got a lot of error messages, how can I do it?
1.     Reference is correct 
2.     Please assist me to tell me how to do it if the reference isn’t correct 
 
 
 
[How to configure to allow Swagger URL ? · Issue #4453 · ant-media/Ant-Media-Server (github.com)](https://github.com/ant-media/Ant-Media-Server/issues/4453)
 
1.  git clone https://github.com/ant-media/Ant-Media-Server.git
2.  Change `host` url to `BroadcastRestService.java` file: `https://github.com/ant-media/Ant-Media-Server/blob/master/src/main/java/io/antmedia/rest/BroadcastRestService.java#L69`
3.  cd Ant-Media-Server; 
4.  mvn compile com.github.kongchen:swagger-maven-plugin:generate;
 
 
I got a lot of errors when I run mvn compile com.github.kongchen:swagger-maven-plugin:generate;
![image](https://github.com/ant-media/Ant-Media-Server/assets/15116422/52e89229-0f6e-48c2-ae0d-36faec7a8864)
